web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v445/DoiServerDatabaseMigration.java

```diff
@@ -0,0 +1,150 @@
+                    doiKey = resultSet.getString(columnForValue);
+                } else if (resultSet.getString(columnForName).equalsIgnoreCase("system/publication/doi/doilandingpagetemplate")) {
+                    doiLandingPageTemplate = resultSet.getString(columnForValue);
+                } else if (resultSet.getString(columnForName).equalsIgnoreCase("system/publication/doi/doipublicurl")) {
+                    doiPublicUrl = resultSet.getString(columnForValue);
+                } else if (resultSet.getString(columnForName).equalsIgnoreCase("system/publication/doi/doipattern")) {
+                    doiPattern = resultSet.getString(columnForValue);
+                }
+
+            }
+        }
+
+        if (doiEnabled) {
+
+            // Check the information is filled
+            boolean createDoiServer = StringUtils.hasLength(doiUrl) &&
+                StringUtils.hasLength(doiUsername) &&
+                StringUtils.hasLength(doiPassword) &&
+                StringUtils.hasLength(doiKey) &&
+                StringUtils.hasLength(doiPattern);
+
+            if (createDoiServer) {
+                try (PreparedStatement update = connection.prepareStatement(
+                    "INSERT INTO doiservers " +
+                        "(id, isdefault, landingpagetemplate, name, url, username, password, pattern, prefix, publicurl) " +
+                        "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)")
+                ) {
+
+                    update.setInt(1, 1);
+                    update.setString(2, "y");
+                    update.setString(3, doiLandingPageTemplate);
+                    update.setString(4, "Default DOI server");
+                    update.setString(5, doiUrl);
+                    update.setString(6, doiUsername);
+                    update.setString(7, doiPassword);
+                    update.setString(8, doiPattern);
+                    update.setString(9, doiKey);
+                    update.setString(10, doiPublicUrl);
+
+                    update.execute();
+
+                } catch (java.sql.BatchUpdateException e) {
+                    connection.rollback();
+                    Log.error(Geonet.GEONETWORK, "Error occurred while creating the DOI server:" + e.getMessage(), e);
+                    SQLException next = e.getNextException();
+                    while (next != null) {
+                        Log.error(Geonet.GEONETWORK, "Next error: " + next.getMessage(), next);
+                        next = e.getNextException();
+                    }
+
+                    throw new RuntimeException(e);
+                } catch (Exception e) {
+                    connection.rollback();
+
+                    throw new Error(e);
+                }
+
+
+                try (PreparedStatement delete = connection.prepareStatement(
+                    "DELETE FROM Settings WHERE name LIKE 'system/publication/doi%'")
```

Comment:
We still need the DOI settings to enable it or not. 

```suggestion
                    "DELETE FROM Settings WHERE name LIKE 'system/publication/doi%' and name != 'system/publication/doi/doienabled'")
```

It is used on the client app eg. https://github.com/geonetwork/core-geonetwork/blob/74cdf4ab36536e052bccff064d2b97915e302f89/web-ui/src/main/resources/catalog/components/doi/DoiService.js#L45